lib/fs.js

```diff
@@ -1097,7 +1097,9 @@ function writeAll(fd, buffer, offset, length, position, callback) {
       } else {
         offset += written;
         length -= written;
-        position += written;
+        if (position !== null) {
```

Comment:
I don't see how `position` can become `undefined` or `false` in any of the two code paths.